crates/orchestrator/src/tests/jobs/state_update_job/mod.rs

```diff
@@ -0,0 +1,76 @@
+use mockall::predicate::eq;
+use rstest::*;
+use settlement_client_interface::MockSettlementClient;
+
+use std::{collections::HashMap, fs};
+
+use super::super::common::init_config;
+
+use crate::jobs::{
+    constants::JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY,
+    state_update_job::StateUpdateJob,
+    types::{JobStatus, JobType},
+    Job,
+};
+
+use httpmock::prelude::*;
+
+#[rstest]
+#[tokio::test]
+async fn test_create_job() {
+    let config = init_config(None, None, None, None, None, None).await;
+
+    let job = StateUpdateJob.create_job(&config, String::from("0"), HashMap::default()).await;
+    assert!(job.is_ok());
+
+    let job = job.unwrap();
+
+    let job_type = job.job_type;
+    assert_eq!(job_type, JobType::StateTransition, "job_type should be StateTransition");
+    assert!(!(job.id.is_nil()), "id should not be nil");
+    assert_eq!(job.status, JobStatus::Created, "status should be Created");
+    assert_eq!(job.version, 0_i32, "version should be 0");
+    assert_eq!(job.external_id.unwrap_string().unwrap(), String::new(), "external_id should be empty string");
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_process_job() {
```

Comment:
let's also add test cases where it should fail if block numbers aren't increasing?

Reply:
Added!